mob/src/main/java/com/github/phantazmnetwork/mob/spawner/BasicMobSpawner.java

```diff
@@ -3,64 +3,169 @@
         this.mobStore = Objects.requireNonNull(mobStore, "mobStore");
         this.neuralSpawner = Objects.requireNonNull(neuralSpawner, "neuralSpawner");
+        this.contextManager = Objects.requireNonNull(contextManager, "contextManager");
+        this.keyParser = Objects.requireNonNull(keyParser, "keyParser");
     }
 
     @Override
     public @NotNull PhantazmMob spawn(@NotNull Instance instance, @NotNull Point point, @NotNull MobModel model) {
         NeuralEntity neuralEntity = neuralSpawner.spawnEntity(instance, point, model.getDescriptor());
-        setDisplayName(neuralEntity, model);
+        setEntityMeta(neuralEntity, model);
         setEquipment(neuralEntity, model);
         setAttributes(neuralEntity, model);
         setHealth(neuralEntity, model);
 
-        PhantazmMob mob = new PhantazmMob(model, neuralEntity);
-        addGoals(mob);
+        Module module = new Module(neuralEntity);
+        DependencyProvider dependencyProvider = new ModuleDependencyProvider(module, keyParser);
+        ElementContext context = contextManager.makeContext(model.getNode());
+        addGoals(context, dependencyProvider, neuralEntity, model);
+        Map<Key, Collection<Skill>> triggers = createTriggers(context, dependencyProvider, neuralEntity, model);
 
+        PhantazmMob mob = new PhantazmMob(model, neuralEntity, triggers);
         mobStore.registerMob(mob);
         return mob;
     }
 
-    private void setDisplayName(@NotNull NeuralEntity neuralEntity, @NotNull MobModel model) {
-        model.getDisplayName().ifPresent(displayName -> {
-            neuralEntity.setCustomName(displayName);
-            neuralEntity.setCustomNameVisible(true);
-        });
+    private void setEntityMeta(@NotNull NeuralEntity neuralEntity, @NotNull MobModel model) {
+        EntityMeta meta = neuralEntity.getEntityMeta();
+        ConfigNode metaNode = model.getMetaNode();
+        for (Method method : meta.getClass().getMethods()) {
+            if (!Modifier.isPublic(method.getModifiers()) || method.getReturnType() != void.class) {
+                continue;
+            }
+
+            Parameter[] parameters = method.getParameters();
+            if (parameters.length != 1) {
+                continue;
+            }
+
+            String methodName = method.getName();
+            if (!methodName.startsWith("set")) {
+                continue;
+            }
+            String key;
+            if (methodName.length() > 3) {
+                key = Character.toLowerCase(methodName.charAt(3)) + methodName.substring(4);
+            }
+            else {
+                key = methodName.substring(3);
+            }
+
+            ConfigElement element = metaNode.getElementOrDefault((ConfigElement)null, key);
+            if (element == null) {
+                continue;
+            }
+
+            Parameter parameter = parameters[0];
+            NotNull[] notNulls = parameter.getAnnotationsByType(NotNull.class);
```

Comment:
NotNull is not a @Repeatable (i probably just mentioned another account) annotation so there will never be more than 1, you can just use getAnnotationByType